merkle/smt/writer_test.go

```diff
@@ -207,15 +244,38 @@ func genUpd(key, value string) NodeUpdate {
 	return NodeUpdate{ID: tree.NewNodeID2(string(key256[:]), 256), Hash: hash}
 }
 
-type noopAccessor struct {
-	get, set error
+// testAccessor implements NodeBatchAccessor for testing purposes.
+type testAccessor struct {
+	mu   sync.RWMutex // Guards the h map.
+	h    map[tree.NodeID2][]byte
+	save bool  // Persist node updates in this accessor.
+	get  error // The error returned by Get.
+	set  error // The error returned by Set.
 }
 
-func (n noopAccessor) Get(context.Context, []tree.NodeID2) (map[tree.NodeID2][]byte, error) {
-	if err := n.get; err != nil {
+func (t *testAccessor) Get(ctx context.Context, ids []tree.NodeID2) (map[tree.NodeID2][]byte, error) {
+	if err := t.get; err != nil || !t.save {
```

Comment:
I applaud the code golf compactness of this, but I implore you to expand `if !t.save { return nil, nil }` as a separate check. Seeing `return nil, err` and the `err` being `nil` seems too clever for its own good. (passim)

Reply:
Done. I cry.